### DISCUSSION TITLE
HoneyGain Infite Restart Loop With Proxies

### DISCUSSION THREAD

**Discussion by ByQwank** [engageub/InternetIncome#127] [Q&A]
HoneyGain container constantly restarting, and won't launch.
Every other container works; besides HoneyGain which goes on a infinite restarting loop.
Tried switching proxies to Socks5:// and HTTP:// same result.
Works for other containers, but not HoneyGain.
Have tried Test & Main Branch on two different computers; same problems arise.

**Comment by engageub** [comment]
Hello,
In that case, it may be your password issue.
Can you enable logs in properties.conf and see what the issue is by checking the docker logs.
In the containernames.txt file you will find container name starting with honey.
Copy that and run the following command. In the following command replace YOUR_HONEYGAIN_CONTAINER_NAME with your container name.

```
sudo docker container logs YOUR_HONEYGAIN_CONTAINER_NAME
```

Are you using double quotes in properties.conf file while using the password. Please use single quotes if your password has special characters
Does your password already have single or double quotes or special characters like $ ?
Are you able to login to honeygain manually through browser?

Thank you

**Comment by engageub** [comment]
You can also run the following command to see if the environment variables has the correct password.

```
sudo docker inspect YOUR_HONEYGAIN_CONTAINER_NAME
```
The above command shows what username and password has been given to the container.

Thank you

**Comment by engageub** [comment]
If your password has special characters like $, please use singe quotes to enclose instead of double quotes.
If you are using double quotes, you will have use backslash (\\) before ($).
Make sure your number of devices in your dashboard has not reached 10 already. 

Thank you

**Comment by nkciam** [comment]
Hi @engageub , I have same issue. Here are the logs of honeygain container. 
![Capture](https://github.com/engageub/InternetIncome/assets/70848927/34063dbe-d7f0-4d5b-aeb9-94671577acdb)
My honeygain password does not have a special character. Simple "abcde199x".  Tried with both "" and '' in the password field. HG container still continuous restart!

**Reply by engageub** [reply]
Hi,
Could you please let me know if you are using amd64 architecture or arm64 architecture.



Thank you

**Reply by nkciam** [reply]
Hi, I'm using arm64 architecture :) Does HG support it?

**Reply by engageub** [reply]
Have you installed the following on your PC which is mentioned in https://github.com/engageub/InternetIncome
for ARM64 architecture. 

```
sudo docker run --privileged --rm tonistiigi/binfmt --install all
sudo apt-get install qemu binfmt-support qemu-user-static
```

**Reply by engageub** [reply]
If you are using Oracle free VPS, it provides two AMD64 instances with 1 GB RAM each. You can run honeygain on those instances.

**Comment by engageub** [comment]
If you are using proxy, you can try that on [play with docker](https://labs.play-with-docker.com/). It is working fine there. You get 5 free instances every 4 hours. 
Please use [tunproxy](https://github.com/engageub/InternetIncome/tree/tunproxy) branch on play with docker instead of main branch.
If its working there, but not on your PC, then it is likely that some files are not downloaded or there is compatibility issue. 

Thank you

**Reply by engageub** [reply]
For proxies, the script is using older version of honeygain which is amd64 compatible. This version is confirmed to be working fine with proxies. 
Direct connection uses latest version which supports multiple platforms. 
If you would like to try the latest version on proxies,  please replace line 584 in internetIncome.sh file with the following instead of older version 0.6.6.

```
 honeygain_image="honeygain/honeygain"
```

Thank you

**Reply by nkciam** [reply]
Work like a charm with this image "honeygain_image="honeygain/honeygain"". Thank a lot! Have a good weekend sir!

**Reply by engageub** [reply]
Thank you

> Work like a charm with this image "honeygain_image="honeygain/honeygain"". Thank a lot! Have a good weekend sir!

Thank you for the confirmation. Do you see the devices online with proxy. Are you using proxies with Socks5 UDP support.

**Reply by nkciam** [reply]
![Capture](https://github.com/engageub/InternetIncome/assets/70848927/46f1be36-ee90-4da6-9370-f118da9942fd)
Yes, my proxy worked and online. It's waiting for traffic like this picture. My proxy not Socks5 UDP support. I'm using US Residential proxy with HTTP type.

**Reply by engageub** [reply]
This was the issue for which older version was used and was confirmed to be working. However, sometimes if the IP is not considered residential, that can happen too. 
If it shows "Connected. Waiting for traffic.."  for longer time, you may use the older version or try using socks5 proxy which supports UDP.

Thank you

**Reply by nkciam** [reply]
For using older version 0.6.6. Could I change the image to arm like this? "honeygain_image="--arm64 honeygain/honeygain:0.6.6"".

**Reply by nkciam** [reply]
![Screenshot_20240601-013708_Chrome](https://github.com/engageub/InternetIncome/assets/70848927/92698579-a7bf-40fe-9f89-a85a81deaaca)
Seem HG support ARM architecture from 0.8.1 image tag version
https://support.honeygain.com/hc/en-us/articles/360018979919-How-to-run-Honeygain-on-Docker-Linux

**Reply by engageub** [reply]
> For using older version 0.6.6. Could I change the image to arm like this? "honeygain_image="--arm64 honeygain/honeygain:0.6.6"".

No. You have to set the value of line 584 as it was earlier in the main branch. It only supports amd64. 
You have to install emulation layer on arm64 for it to work. Have you installed the tools mentioned above for arm64 which makes it compatible to run amd64 images. 


Thank you

**Reply by engageub** [reply]
Yes, the latest version supports multiple platforms. But with proxies it was not working. So try to use proxy which supports UDP.  With direct connection, it is using the latest version only. 

Thank you

**Reply by nkciam** [reply]
Thanks! Let's me find proxy site sell residential support UDP and test it again. I will report the test later.
